packages/arb-token-bridge-ui/src/components/TransferPanel/TokenSearch.tsx

```diff
@@ -144,6 +143,16 @@ function TokensPanel({
   const tokensFromUser = useTokensFromUser()
   const tokensFromLists = useTokensFromLists()
 
+  const tokensL2AddressOnly = useMemo(() => {
+    const allTokens = [
+      ...new Set([
+        ...Object.values(tokensFromUser),
+        ...Object.values(tokensFromLists)
+      ])
+    ]
+    return allTokens.map(token => token?.l2Address)
```

Comment:
Seems like we only return lowercased tokens from lists, but do we have the same guarantee that tokens from users are going to be all lowercase?

In `addNewToken` we don't lowercase it either, just sanity checking here.

Reply:
good point